src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -309,5 +310,22 @@ static INamedTypeSymbol UnwrapNullableValueTypes(INamedTypeSymbol typeSymbol)
                 return typeSymbol;
             }
         }
+
+        private static bool IsValidParseCall(IInvocationOperation invocationOperation, INamedTypeSymbol? guidType)
+        {
+            var targetMethod = invocationOperation.TargetMethod;
+
+            if (targetMethod.Name != "Parse")
```

Comment:
NOTE: We ideally don't recommend comparing method names to identify if the expected method is being called. Instead you should fetch the expected method symbol on the `guidType` named type symbol in the compilation start action and directly compare `invocationOperation.TargetMethod` and that well-known method symbol. However, I see that this analyzer already follows the pattern similar to the one you followed here, so I won't block your PR on this: https://github.com/dotnet/roslyn-analyzers/blob/a4ffd4f27aa2847deecad09d87b8bd4c8c301a0c/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs#L267-L275

I'll open a tracking issue to clean this up in a follow-up PR.

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/6052